### PR TITLE
Index pidx load

### PR DESCRIPTION
**Pull request by bgn42**
## Fixes

-

## Changes

Automagically load of index.pidx
- 355, 357, 358

## Checklist

- [x] 🤖 This change is covered by unit tests as required.
- [x] 🤹 All required manual testing has been performed.
- [x] 🛡️ Security impacts have been considered.
- [ ] 📖 All documentation updates are complete.
- [x] 🧠 This change does not change third-party dependencies

### CODE REVIEW

File: cmd/commands/init_test.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ var initCmdTests = []TestCase{
 		name:           "test create using local index.pidx that do not exist",
```
Comment (jkrech) [inline cmd/commands/init_test.go:57]: that does not exist

File: cmd/errors/errors.go [hunk 2]
```diff
@@ -54,6 +54,7 @@ var (
 	ErrFailedInflatingFile       = errors.New("fail to inflate file")
 	ErrFailedCreatingDirectory   = errors.New("fail to create directory")
 	ErrFileNotFound              = errors.New("file not found")
+	ErrFileNotFoundUseInit       = errors.New("index.pdix file not found; use cpackget init command to retrieve it")
```
Comment (jkrech) [inline cmd/errors/errors.go:57]: spelling:  index.pidx

File: cmd/errors/errors.go [hunk 3]
```diff
@@ -95,4 +96,7 @@ var (
 
 	// Error/Flag to detect when a user has requested early termination
 	ErrTerminatedByUser = errors.New("terminated by user request")
+
+	ErrIndexTooOld = errors.New("public index too old")
```
Comment (jkrech) [inline cmd/errors/errors.go:100]: Is the public index synonym for "index.pidx", maybe we just use the filename everywhere?

File: cmd/errors/errors.go [hunk 4]
```diff
@@ -95,4 +96,7 @@ var (
...
+	ErrOffline     = errors.New("remote server is offline or cannot be reached")
```
Comment (jkrech) [inline cmd/errors/errors.go:101]: "remote server" - can we provide a url here?

File: cmd/installer/root.go [hunk 5]
```diff
@@ -1075,7 +1078,7 @@ func SetPackRoot(packRoot string, create bool) error {
 		return errs.ErrPackRootDoesNotExist
 	}
 
-	checkConnection := viper.GetBool("check-connection")
+	checkConnection := viper.GetBool("check-connection") // TODO: never set
```
Comment (jkrech) [inline cmd/installer/root.go:1081]: What does the TODO has as implications?

Comment (bgn42) [reply cmd/installer/root.go:1081]: TODO here means to find the reason why this command line option was there someday but then removed. Also it means to look deeper what this option was thought to do. A quick look does not clearly show what it was good for.

File: cmd/commands/update_index.go [hunk 6]
```diff
@@ -43,7 +43,7 @@ var UpdateIndexCmd = &cobra.Command{
 }
 
 func getLongUpdateDescription() string {
-	return `Updates the public index in ` + os.Getenv("CMSIS_PACK_ROOT") + `/.Web/index.pidx using the URL in <url> tag inside index.pidx.
+	return `Updates the public index in ` + os.Getenv("CMSIS_PACK_ROOT") + "/.web/" + installer.PublicIndex + " using the URL in <url> tag inside " + installer.PublicIndex + `.
```
Comment (jkrech) [inline cmd/commands/update_index.go:46]: Was passiert hier, wenn CMSIS_PACK_ROOT nicht im User Environment gesetzt ist. Setzt cpackget das auf default?

**Review by jkrech — APPROVED** [review]
LGTM

File: cmd/commands/update_index.go (hunk 6 above)
Comment (bgn42) [reply cmd/commands/update_index.go:46]: Da ist dann nichts. Muss ich wohl vorher abfragen. Aber ist nur für die Help-Message...

**Review by jkrech — APPROVED** [review]
LGTM

File: cmd/commands/update_index.go (hunk 6 above)
Comment (jkrech) [reply cmd/commands/update_index.go]: If CMSIS_PACK_ROOT is not set, than cpackget uses the host specific default locations like e.g. on windows:
"c:\users\<user>\AppData\Local\Arm\Packs" or if `-R <dir>` is specified then that location is used. 
I am pretty certain that there must be a variable representing this location for the running session.

**Review by jkrech — APPROVED** [review]
LGTM

**Pull request merged**